github/github.go

```diff
@@ -397,14 +398,22 @@ func (c *Client) Rate() Rate {
 // first decode it.  If rate limit is exceeded and reset time is in the future,
 // Do returns *RateLimitError immediately without making a network API call.
 func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
+	return c.do(context.TODO(), req, v)
+}
+
+// TODO: The signature of Do is kept unmodified temporarily, in order for tests
+//       to pass during the transition period. Once all endpoints are updated
+//       to accept ctx parameter, the old Do method can be removed, and this
+//       new do method with ctx parameter should be made to take its place.
+func (c *Client) do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
 	rateLimitCategory := category(req.URL.Path)
```

Comment:
I think it's unnecessary.

The `context` package is already [documented](https://godoc.org/context) to say that `nil` value of `context.Context` shouldn't be passed:

> Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use.

So all valid code will always have non-nil.

Invalid code can be easily caught by static analysis tools such as go vet, as the `context` package docs allude:

> Programs that use Contexts should follow these rules to keep interfaces consistent across packages and enable static analysis tools to check context propagation

> TODO is recognized by static analysis tools that determine whether Contexts are propagated correctly in a program.